src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -0,0 +1,24 @@
+<%--
+  Copyright 2017 Google Inc.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--%>
+<%-- still trying to get the hang of jsp, not sure if this is correctly implemented --%>
+<%  List<String> friends_usernames = Arrays.aslist('Ricardo', 'Elona', 'Drew');
+    String username = (String) request.getSession().getAtrribute('user');
```

Comment:
You should pass this information to the JSP file via the request or response object in the servlet. The more logic we keep in the Servlet, the more we can unit test and the more errors the compiler can catch.

Check the ChatServlet to see how it passes the conversation and messages to JSP :)